### PR TITLE
The annotation currently being edited is now clearly identifiable

### PR DESCRIPTION
**Pull request by kerryland**
When editing annotations it's now each to tell which piece of text is related to the comment:

image

A couple of notes:
- My change to `EditAnnotations::StartEditAnnotations` was simply because I was offended by the repetition of calls to `SetColor` :-)
- `EditAnnotations::SelectAnnotationInEditWindow` has a new call to `RepaintAsync`, which smells a bit to me. There might be a better way to achieve the same goal, but I don't know it.

Solves most of my problem raised in [issue 3638](https://github.com/sumatrapdfreader/sumatrapdf/issues/3638), somebody else's in [issue 3590](https://github.com/sumatrapdfreader/sumatrapdf/issues/3590), and [issue 1996](https://github.com/sumatrapdfreader/sumatrapdf/issues/1996)

PS: Sorry about the additional commits in this pull request that are actually just dumb merges from master, my git skills are rusty, but I don't think it matters.

### CODE REVIEW

**Comment by kerryland** [comment]
I think this also fixes https://github.com/sumatrapdfreader/sumatrapdf/issues/1688

File: src/EditAnnotations.cpp [hunk 1]
```diff
@@ -1535,26 +1549,40 @@ Annotation* EngineMupdfCreateAnnotation(EngineBase* engine, AnnotationType typ,
 
     pdf_update_annot(ctx, annot);
     auto res = MakeAnnotationPdf(epdf, annot, pageNo);
+
+    std::optional <PdfColor> col;
```
Comment (kjk) [inline src/EditAnnotations.cpp:1553]: We don't use std::, please use ColorUnset instead

Comment (kerryland) [reply src/EditAnnotations.cpp]: Fixed.

**Comment by kjk** [comment]
Thanks!

**Comment by kerryland** [comment]
FYI there are some minor bugs that I need to fix:
1. Navigation via the `ctrl-dblclick` function doesn't update the highlighted annotation immediately
2. Navigation to `CmdEditAnnotations` via a keyboard shortcut doesn't update the highlighted annotation

**Pull request merged**